test/e2e/framework/service/jig.go

```diff
@@ -55,9 +57,31 @@ import (
 // NodePortRange should match whatever the default/configured range is
 var NodePortRange = utilnet.PortRange{Base: 30000, Size: 2768}
 
-// It is copied from "k8s.io/kubernetes/pkg/registry/core/service/portallocator"
 var errAllocated = errors.New("provided port is already allocated")
 
+type staticPortRange struct {
+	sync.Mutex
+	baseport      int
+	length        int
+	reservedPorts sets.Set[int]
+}
+
```

Comment:
add reference to the KEP to understand where this algorithm is coming https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3668-reserved-service-nodeport-range